Sources/Workspace/Diagnostics.swift

```diff
@@ -42,3 +42,158 @@ extension ManifestParseError: DiagnosticDataConvertible {
         }
     }
 }
+
+extension Workspace {
```

Comment:
How about defining the diagnostics under an enum called `WorskpaceDiagnostics`, that way we don't need add the suffix "Diagnostic" in each struct. e.g.: `WorskpaceDiagnostics.UncommitedChanges`